src/classes/SftpSessionHandler.ts

```diff
@@ -1108,194 +1134,180 @@ export class SftpSessionHandler {
-      // These flags are explained in the NodeJS fs documentation:
-      // https://nodejs.org/api/fs.html#file-system-flags
-      switch (flagsString) {
-        case 'w': // write
-        case 'wx': // write (file must not exist)
-        case 'xw': // write (file must not exist)
-        case 'w+': // write and read
-        case 'xw+': // write and read (file must not exist)
-        case 'ax': // append (file must not exist)
-        case 'xa': // append (file must not exist)
-        case 'a+': // append and read
-        case 'ax+': // append and read (file must not exist)
-        case 'xa+': // append and read (file must not exist)
-        case 'a': // append
-        {
-          this.temporaryFileManager.createTemporaryFile(filePath).then(() => {
-            const temporaryFileResource = {
-              resourceType: ServerResourceType.TemporaryFile as const,
-              virtualFilePath: filePath,
-            };
-            this.activeHandles.set(handle, temporaryFileResource);
-            logger.verbose(
-              'Response: Handle',
-              {
-                reqId,
-                handle,
-                path: filePath,
-              },
-            );
-            this.sftpConnection.handle(
-              reqId,
-              Buffer.from(handle),
-            );
-          }).catch((err) => {
-            logger.debug(err);
-            logger.verbose(
-              'Response: Status (FAILURE)',
-              {
-                reqId,
-                code: SFTP_STATUS_CODE.FAILURE,
-              },
-            );
-            this.sftpConnection.status(
-              reqId,
-              SFTP_STATUS_CODE.FAILURE,
-              'An error occurred when attempting to create the file in temporary storage.',
-            );
-          });
-          break;
+
+    (async () => {
+      try {
+        await permanentFileSystem.loadDirectory(filePath);
+        logger.verbose(
+          'Response: Status (FAILURE)',
+          {
+            reqId,
+            code: SFTP_STATUS_CODE.FAILURE,
+          },
+        );
+        this.sftpConnection.status(
+          reqId,
+          SFTP_STATUS_CODE.FAILURE,
+          'This path is a folder, so it cannot be opened as a file.',
+        );
+        return;
+      } catch (err) {
+        if (err instanceof ResourceDoesNotExistError) {
+          // This is what we want -- the folder shouldn't exist.
+        } else {
+          throw err;
         }
-        case 'r+': // read and write (error if doesn't exist)
-        case 'r': // read
-        default:
-          logger.verbose(
-            'Response: Status (NO_SUCH_FILE)',
-            {
-              reqId,
-              code: SFTP_STATUS_CODE.NO_SUCH_FILE,
-              path: filePath,
-            },
-          );
-          this.sftpConnection.status(
+      }
+
+      try {
+        const file = await permanentFileSystem.loadFile(filePath, true);
```

Comment:
This should possibly NOT pass `true` in the second parameter -- I think we want to load the file using cached data if cached data exists.